data-prepper-plugins/otel-logs-source/src/main/java/org/opensearch/dataprepper/plugins/source/otellogs/http/ArmeriaHttpService.java

```diff
@@ -0,0 +1,103 @@
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.otel.codec.OTelProtoCodec;
+import org.opensearch.dataprepper.plugins.otel.codec.OTelProtoOpensearchCodec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.annotation.Consumes;
+import com.linecorp.armeria.server.annotation.Post;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.Timer;
+import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceRequest;
+import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceResponse;
+
+public class ArmeriaHttpService {
+    private static final Logger LOG = LoggerFactory.getLogger(ArmeriaHttpService.class);
+
+    public static final String REQUESTS_RECEIVED = "requestsReceived";
+    public static final String SUCCESS_REQUESTS = "successRequests";
+    public static final String PAYLOAD_SIZE = "payloadSize";
+    public static final String REQUEST_PROCESS_DURATION = "requestProcessDuration";
+
+    private final OTelProtoCodec.OTelProtoDecoder oTelProtoDecoder;
+    private final Buffer<Record<Object>> buffer;
+
+    private final int bufferWriteTimeoutInMillis;
+
+    private final Counter requestsReceivedCounter;
+    private final Counter successRequestsCounter;
+    private final DistributionSummary payloadSizeSummary;
+    private final Timer requestProcessDuration;
+
+    public ArmeriaHttpService(Buffer<Record<Object>> buffer, final PluginMetrics pluginMetrics, final int bufferWriteTimeoutInMillis) {
+        this.buffer = buffer;
+        this.oTelProtoDecoder = new OTelProtoOpensearchCodec.OTelProtoDecoder();
+        this.bufferWriteTimeoutInMillis = bufferWriteTimeoutInMillis;
+
+        requestsReceivedCounter = pluginMetrics.counter(REQUESTS_RECEIVED);
+        successRequestsCounter = pluginMetrics.counter(SUCCESS_REQUESTS);
+        payloadSizeSummary = pluginMetrics.summary(PAYLOAD_SIZE);
+        requestProcessDuration = pluginMetrics.timer(REQUEST_PROCESS_DURATION);
+    }
+
+    // no path provided. Will be set by config.
+    @Post("")
+    @Consumes(value = "application/json")
+    public ExportLogsServiceResponse exportLog(ExportLogsServiceRequest request) {
+        requestsReceivedCounter.increment();
+        payloadSizeSummary.record(request.getSerializedSize());
+
+        requestProcessDuration.record(() -> processRequest(request));
+
+        return ExportLogsServiceResponse.newBuilder().build();
+    }
+
+    private void processRequest(final ExportLogsServiceRequest request) {
+        final List<OpenTelemetryLog> logs;
+
+        try {
+            logs = oTelProtoDecoder.parseExportLogsServiceRequest(request, Instant.now());
+        } catch (Exception e) {
+            LOG.error("Failed to parse the request {} due to:", request, e);
```

Comment:
Please use warning level and sensitive request marker.
```suggestion
            LOG.warn(DataPrepperMarkers.SENSITIVE, "Failed to parse request with error '{}'. Request body: {}.", e.getMessage(), request);
```

Reply:
done.

Is this a general rule? Use WARN-level with SENSITIVE-Marker for errors? 

Is there any documentation on how to logging should be done in DataPrepper? Could not find any in the README